File: src/main/java/org/nomad/pithos/models/GameObject.java
package org.nomad.pithos.models;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.Id;
import java.io.Serializable;

@Data
@Entity
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class GameObject implements Serializable {
    @Id
    private String id;
    private long creationTime;
    private long lastModified;
    private long ttl;
    private byte[] value;
}

File: src/main/java/org/nomad/pithos/models/GameObject.kt
package org.nomad.pithos.models

import java.io.Serializable
import javax.persistence.Entity
import javax.persistence.Id

@Entity
data class GameObject(@Id val id: String, val creationTime: Long, val lastModified: Long, val ttl: Long, val value: ByteArray) : Serializable {
}